File: Algo/hr_implement/4/equalizeArray.js
function equalizeArray(a) {
  let max = 0,
    hashMap = new Map();
  for (let i = 0; i < a.length; i++) {
    let val = 1;
    if (hashMap.has(a[i])) {
      val = hashMap.get(a[i]) + 1;
      hashMap.delete(a[i]);
    }
    hashMap.set(a[i], val);
    if (val > max) {
      max = val;
    }
  }
  return a.length - max;
}

let a = [3, 3, 1, 2, 3];
console.log('ans: ', equalizeArray(a));

File: Algo/hr_implement/4/equalizeArray.cpp
// Print the number of letter a's in the first n letters of infinite string

#include <iostream>
#include <vector>
#include <map>
using namespace std;

int equalizeArray(vector<int> arr)
{
    map<int, int> m;
    int max = 0;
    for (int i = 0; i < arr.size(); i++)
    {
        m[arr[i]]++;
        if (m[arr[i]] > max)
            max = m[arr[i]];
    }

    return arr.size() - max;
}

int main()
{
    vector<int> arr(5);
    arr[0] = 3;
    arr[1] = 3;
    arr[2] = 2;
    arr[3] = 1;
    arr[4] = 3;

    cout << "Ans: " << equalizeArray(arr) << endl;
    return 0;
}
